### PR TITLE
Do not publish the image in PR builds.

### PR DESCRIPTION
**Pull request by timuralp**
On PR builds, we should not try to publish the image to docker hub.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,17 +1,15 @@
...
-  - export TAG=`if [ "$TRAVIS_BRANCH" == "master" ]; then echo "latest"; else echo $TRAVIS_BRANCH ; fi`
+
+script:
   - docker build -f Dockerfile -t $REPO:$COMMIT .
-  - docker tag $REPO:$COMMIT $REPO:$TAG
-  - docker tag $REPO:$COMMIT $REPO:travis-$TRAVIS_BUILD_NUMBER
-  - docker push $REPO
+
+after_success:
+  - 'if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then bash ./travis/.publish-image.sh; fi'
```
[the remark this answers is not in the document]
Comment (timuralp) [reply .travis.yml]: Nope -- good catch!

File: .publish_img.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
```
Comment (gaul) [inline .publish_img.sh]: Why name this script with a leading dot?

File: .travis.yml [hunk 3]
```diff
@@ -1,17 +1,15 @@
...
-  - export TAG=`if [ "$TRAVIS_BRANCH" == "master" ]; then echo "latest"; else echo $TRAVIS_BRANCH ; fi`
+
+script:
   - docker build -f Dockerfile -t $REPO:$COMMIT .
-  - docker tag $REPO:$COMMIT $REPO:$TAG
-  - docker tag $REPO:$COMMIT $REPO:travis-$TRAVIS_BUILD_NUMBER
-  - docker push $REPO
+
+after_success:
+  - 'if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then bash $TRAVIS_BUILD_DIR/.publish_img.sh; fi'
```
Comment (gaul) [inline .travis.yml]: Make executable instead of explicitly invoking bash?

Comment (gaul) [inline .travis.yml]: Inverted comparison.

**Pull request merged**